### DISCUSSION TITLE
Question - How to design so that I can feed language grammar as part of conversation?

### DISCUSSION THREAD

**Discussion by levirtevs** [winkjs/wink-nlp#101] [General]
![How](https://user-images.githubusercontent.com/112684060/207354619-42d8de22-1480-46fb-8bf5-0eb69ce16248.png)

**Comment by sanjayaksaxena** [comment]
Hello @levirtevs 

A question answering system can get very complex. At one end you have a simple system close-ended system capable of handling limited questions from a specific domain to complex systems like ChatGPT. Depending on the requirement one can create a simple rule-based system to a ML driven one involving large corpus & training steps.

Assuming you have a closed-ended system with a small corpus from where the answers have to be extracted, you can experiment with the following simple approach:

1. Based on the question, use a [BM25 based](https://winkjs.org/wink-nlp/bm25-vectorizer.html) [search](https://github.com/winkjs/showcase-bm25-text-search) to first identify document(s) that are likely to contain the answer.
2. From the selected identify the highest scoring [sentence](https://winkjs.org/wink-nlp/sentences.html) in context of the question using [similarity](https://winkjs.org/wink-nlp/similarity.html) methods; start with a simple BoW model... better efficacy may be achieved using word vectors.

Beyond this, please refer to recent research on the subject.

Best,
Sanjaya
